bin/common.bash

```diff
@@ -228,3 +228,25 @@ check_openstack_credentials() {
         exit 1
     fi
 }
+
+# Compares the expected and actual git state of the kubespray submodule.
+# If they don't match the user will be asked if they want to continue anyway.
+kubespray_version_check(){
+   pushd "${here}/../" || exit
```

Comment:
I still don't think you need the `exit` command here since we are running with `set -e`

```suggestion
   pushd "${here}/../"
```

Same on the `popd` command

Reply:
check for merge conflicts................................................Passed
check yaml...........................................(no files to check)Skipped
don't commit to branch...................................................Passed
Test shell scripts with shellcheck.......................................Failed
- hook id: shellcheck
- exit code: 1

In bin/common.bash line 235:
   pushd "${here}/../"
   ^-----------------^ SC2164: Use 'pushd ... || exit' or 'pushd ... || return' in case pushd fails.

Did you mean: 
   pushd "${here}/../" || exit


In bin/common.bash line 239:
   popd
   ^--^ SC2164: Use 'popd ... || exit' or 'popd ... || return' in case popd fails.